lib/spack/spack/test/conftest.py

```diff
@@ -1852,3 +1852,24 @@ def _factory(rpaths, message="Hello world!"):
         return executable
 
     return _factory
+
+
+@pytest.fixture(scope="session")
+def concretized_specs_cache():
+    """Cache for mock concrete specs"""
+    return {}
+
+
+@pytest.fixture
+def mock_concretize(config, mock_packages, concretized_specs_cache):
+    """Return a concretized spec from immutable mock configuration."""
+
+    def _func(spec_str, tests=False):
+        key = spec_str, tests
+        if key not in concretized_specs_cache:
+            result = spack.spec.Spec(spec_str).concretized(tests=tests)
+            assert result.concrete, "{} is expected to be concrete".format(spec_str)
+            concretized_specs_cache[key] = result
+        return concretized_specs_cache[key]
```

Comment:
Wouldn't you wanna copy?

Reply:
I was doing it in tests, but given it's fast enough we can do it here and avoid doing it at the calling site.